lib/utils/__tests__/mailup.test.ts

```diff
@@ -4,37 +4,20 @@ import Mail = require("nodemailer/lib/mailer");
 
 import * as nodemailer from "nodemailer";
 
-jest.mock("superagent");
 import * as request from "superagent";
 
 import { MailUpTransport, SmtpAuthInfo } from "../mailup";
 
-interface IResponse {
-  readonly body: {
-    readonly Code: string;
-    readonly Message: string;
-    readonly Status: string;
-  };
-  readonly error?: string;
-  readonly status: number;
-  readonly text?: string;
-}
-
-// get superagent mock instance
-const superagent = request("method", "url");
-
-const setMockResponse = (response: IResponse) => {
-  (superagent as any).__setResponse(response);
-};
-const setResponseSpy = (responseSpy: (() => void)) => {
-  (superagent as any).__setResponseSpy(responseSpy);
-};
-const setRequestSpy = (requestSpy: (() => void)) => {
-  (superagent as any).__setRequestSpy(requestSpy);
-};
-const resetSuperagentMocks = () => {
-  (superagent as any).__resetMocks();
-};
+afterEach(() => {
+  jest.restoreAllMocks();
+  jest.resetAllMocks();
+});
+
+const mockSuperagentResponse = (response: any) =>
+  // tslint:disable-next-line:no-object-mutation
+  (Object.getPrototypeOf(request("", "")).send = jest.fn()).mockReturnValueOnce(
```

Comment:
I can't even understand what this does, please explain it very well :)

Reply:
added docs